ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/polygon/DrawAreaTaskMapFragment.kt

```diff
@@ -19,26 +19,33 @@ import android.os.Bundle
 import androidx.lifecycle.lifecycleScope
 import com.google.android.ground.ui.common.AbstractMapFragmentWithControls
 import com.google.android.ground.ui.common.BaseMapViewModel
+import com.google.android.ground.ui.datacollection.DataCollectionFragment
 import com.google.android.ground.ui.map.CameraPosition
 import com.google.android.ground.ui.map.Feature
 import com.google.android.ground.ui.map.MapFragment
 import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
 import kotlinx.coroutines.launch
 
 @AndroidEntryPoint
-class DrawAreaTaskMapFragment(private val viewModel: DrawAreaTaskViewModel) :
-  AbstractMapFragmentWithControls() {
+class DrawAreaTaskMapFragment @Inject constructor() : AbstractMapFragmentWithControls() {
 
   private lateinit var mapViewModel: BaseMapViewModel
+  private lateinit var viewModel: DrawAreaTaskViewModel
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
+
     mapViewModel = getViewModel(BaseMapViewModel::class.java)
+    val taskId = arguments?.getString("taskId")
+    val dcf = requireParentFragment() as DataCollectionFragment
+    viewModel = dcf.viewModel.getTaskViewModel(taskId!!) as DrawAreaTaskViewModel
```

Comment:
`taskId!!` is something can crash

similar approach https://github.com/google/ground-android/pull/2748 but wrapped up in try catch or maybe some other error. handling approach we need

Reply:
maybe we can have an appraoch, where we don't have to do `dcf.viewModel.getTaskViewModel` for every task

Reply:
How would you recover from taskId not being set in the fragment args? I don't think we can...

Reply:
> 
> maybe we can have an appraoch, where we don't have to do `dcf.viewModel.getTaskViewModel` for every task

What do you suggest?